python/ray/autoscaler/v2/instance_manager/instance_storage.py

```diff
@@ -0,0 +1,198 @@
+import logging
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Set, Tuple
+
+from ray.autoscaler.v2.instance_manager.storage import Storage, StoreStatus
+from ray.core.generated.instance_manager_pb2 import Instance
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class InstanceUpdateEvent:
+    """Notifies the status change of an instance."""
+
+    instance_id: str
+    new_status: int
+
+
+class InstanceUpdatedSuscriber(metaclass=ABCMeta):
+    """Subscribers to instance status changes."""
+
+    @abstractmethod
+    def notify(self, events: List[InstanceUpdateEvent]) -> None:
+        pass
+
+
+class InstanceStorage(object):
```

Comment:
```suggestion
class InstanceStorage:
```

Minor, but it is not needed from python 3